wandb/sdk/artifacts/artifact_file_cache.py

```diff
@@ -38,12 +38,19 @@ def __init__(self, cache_dir: StrPath) -> None:
 
         # NamedTemporaryFile sets the file mode to 600 [1], we reset to the default.
         # [1] https://stackoverflow.com/questions/10541760/can-i-set-the-umask-for-tempfile-namedtemporaryfile-in-python
-        umask_cmd = (sys.executable, "-c", "import os; print(os.umask(22))")
-        umask = int(subprocess.check_output(umask_cmd))
-        self._sys_umask = umask
+        self._sys_umask = self._get_sys_umask_threadsafe()
 
         self._override_cache_path: Optional[StrPath] = None
 
+    @staticmethod
+    def _get_sys_umask_threadsafe() -> int:
```

Comment:
this makes more sense as a top-level function rather than a static method

Reply:
agreed + done